test/helpers/valueSetMapper.js

```diff
@@ -0,0 +1,52 @@
+/**
+ *
+ * @param {Array} valueSetResources array of FHIR ValueSet JSON resources
+ * @returns {Object} valueSetMap for CodeService in cql-execution
+ */
+function mapValueSets(valueSetResources) {
+  const valueSets = {};
+  valueSetResources.forEach((valueSet) => {
+    if (valueSet.compose && valueSet.url) {
+      // Grab id for this valueset (should match FHIR ValueSet url)
+      const valueSetId = valueSet.url;
+      if (!valueSets[valueSetId]) {
+        valueSets[valueSetId] = {};
+      }
+
+      // Grab ValueSet version. This usually is not used.
+      let version = valueSet.version || '';
+      if (version === 'N/A') {
+        version = '';
+      }
+
+      // Create array for valueset members.
+      if (!valueSets[valueSetId][version]) {
+        valueSets[valueSetId][version] = [];
+      }
+
+      // NOTE: This should be using ValueSet.expansion as mentioned above.
+      // Iterate over include components and add all concepts
+      valueSet.compose.include.forEach((include) => {
+        if (include.concept) {
+          include.concept.forEach((concept) => {
+            if (concept.code && include.system) {
+              valueSets[valueSetId][version].push({
+                code: concept.code,
+                system: include.system,
+                version: include.version,
+                display: concept.display,
+              });
+            }
+          });
+        }
+      });
+    } else {
+      // TODO: Handle situation when ValueSet does not have url or compose.
```

Comment:
DM2: No specific plan of action here at this point? I'm assuming we do not have a test case that matches this yet?

Reply:
Nothing specific yet. The mCODE ValueSets as is will never encounter this case. Really I was just guarding for the future here (in case we want to add handling for expansion instead of url or compose or something).

Just really to indicate that this utility technically isn't complete and may change if the valusets for mcode change format at all